source/_integrations/unifiprotect.markdown

```diff
@@ -157,6 +157,34 @@ Your main UniFi Protect NVR device also gets a number of diagnostics sensors tha
 * **Disk Health**: Each disk installed in your NVR will have a disk health sensor. These are simple good/bad sensors and the order is not promised to match the order in UniFi OS. Disk model number is provided as a state attribute though to help map sensor to disk.
 * **Utilization and Storage Sensors**: Several other sensors are also added for uptime, hardware utilization, and distribution details of the video on disk.
 
+## Blueprints
+
+An officially maintained list of blueprints for the UniFi Protect integration can be found on the [Blueprint Exchange on the forums](https://community.home-assistant.io/t/unifi-protect-blueprints/362604). Full details can be found on the forums, but a quick overview of the blueprints are
```

Comment:
That is the recommended way to distribute blueprints as per the docs:

https://www.home-assistant.io/docs/automation/using_blueprints/#finding-new-blueprints
https://www.home-assistant.io/docs/blueprint/tutorial/#share-on-the-blueprint-exchange

When I originally made these, there was no way to bundle them as part of the integration. Otherwise, I would have.

Reply:
No, that is how the community can share blueprints.

Reply:
So, you are saying there is a feature provided _by Home Assistant_ that is _only_ allowed to be used by the "community" and not integration developers? And if an integration developer wants to make "official" blueprints for commonly requested things (like sending a notification when someone rings a doorbell), that is not allowed?

Reply:
> So, you are saying there is a feature provided by Home Assistant that is only allowed to be used by the "community" and not integration developers?

I did not say that?

> And if an integration developer wants to make "official" blueprints for commonly requested things (like sending a notification when someone rings a doorbell), that is not allowed?

That is not an integration feature?

Reply:
> That is not an integration feature?

Indeed, it is. User experience is part of the integration. Even if that extends past the code and into the documentation and automation examples/blueprints. 

Otherwise, you have a question literally every day "how do I send a notification when someone rings my doorbell?" multiple times a week. Or people will just "oh I just use Scrypted because it uses HomeKit Secure Video and automatically send notifications when someone rings the doorbell". 

Documentation is equally as important if not more important as part of the writing the code itself.

> I did not say that?

It is implied. If I am the code owner of an integration and I choose to maintain a list of blueprints for users of common tasks, and you tell me I am not allowed to do that, you are blocking my access to a feature developed by core that I can leverage to benefit and improve the UX of the users of the integration. The blueprints true to open-source nature. They are on Github. OSS Licensed. And you make fork them, make a PR to improve them. 

(p.s. I hope you do not feel like I am trying to attack you or anything, I am just trying to express how unilateral "you cannot do that" decisions without offering alternatives comes off. I really love HA and I just want to make it better)

Reply:
> Indeed, it is. User experience is part of the integration. Even if that extends past the code and into the documentation and automation examples/blueprints.

Currently, Blueprints are not part of integrations. I think it could be a good idea to ship blueprints with integrations, which might be a nice suggestion/implementation or architectural discussion.

> Documentation is equally as important if not more important as part of the writing the code itself.

I agree, I didn't state I was thinking otherwise. 🤷 

> I hope you do not feel like I am trying to attack you or anything

Well, the tone of voice is a bit odd and pushing me out of my comfort zone.
Being an integration maintainer does not give you the right to do whatever you want, we have project standards to follow.

For that reason my review was fairly simple, we do not maintain official blueprints on the community forums, as it is "community" forums. Additionally, Blueprints are not part of the integration at his point.

We are ok with having automation examples, or even links to the community forums that e.g., filter by blueprints with a specific tag.

But stating community blueprints from the official documentation as presented here, that is not something we allow.

Reply:
I'm not sure why you closed the PR at this point, instead of adjusting it. 🤷

Reply:
You said I cannot link to the blueprints. There are no tags in community forums to help make these blueprints stand out. So without being able to link them, it is just "go look for blueprints".

Reply:
> You said I cannot link to the blueprints.

So, I'm not sure if this is a language barrier or something else that is wrong. But I did not say that.

I only said: 

> Home Assistant does not have officially supported/maintained blueprints on the community forums

All the other things at this point are becoming assumptions on that, with some quite offensive responses, which are IMHO absolutely not needed or called for.

Anyways:

We can link to community resources, stating it is community resources.

Another solution is not to host these on the community forums, but, e.g., storing them in the documentation repository and linking them from here (so they become reviewed and version controlled).

I also think, as I said above as well, being able to ship blueprints with integration is a good idea and that you should suggest such a thing and see if we can accomplish such a thing.

Reply:
Alternatively - looking at the importer - it seems that our importer actually supports importing from GitHub repositories. So, maybe we could even ship them with the integration, and link from the docs to import them from the master branch of the core.